File: BasicControlNotification/src/main/java/com/bitmovin/player/samples/notification/basic/MainActivity.java
package com.bitmovin.player.samples.notification.basic;

import android.Manifest;
import android.content.pm.PackageManager;
import android.os.Build;
import android.os.Bundle;

import androidx.activity.result.ActivityResultLauncher;
import androidx.activity.result.contract.ActivityResultContracts;
import androidx.annotation.RequiresApi;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.app.ActivityCompat;
import androidx.core.content.ContextCompat;

import com.bitmovin.player.PlayerView;
import com.bitmovin.player.api.Player;
import com.bitmovin.player.api.source.SourceConfig;
import com.bitmovin.player.ui.notification.PlayerNotificationManager;

public class MainActivity extends AppCompatActivity {
    private static final String NOTIFICATION_CHANNEL_ID = "com.bitmovin.player";
    private static final int NOTIFICATION_ID = 1;

    private PlayerView playerView;
    private Player player;
    private PlayerNotificationManager notificationManager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        playerView = findViewById(R.id.playerView);
        player = playerView.getPlayer();

        // Create a PlayerNotificationManager with the static create method
        // By passing null for the mediaDescriptionAdapter, a DefaultMediaDescriptionAdapter will be used internally.
        notificationManager = PlayerNotificationManager.createWithNotificationChannel(
            this,
            NOTIFICATION_CHANNEL_ID,
            R.string.control_notification_channel,
            NOTIFICATION_ID,
            null
        );
        // Allow to dismiss the Notification
        notificationManager.setOngoing(false);

        // Attach the Player to the PlayerNotificationManager
        notificationManager.setPlayer(player);

        initializePlayer();
    }

    @Override
    protected void onStart() {
        playerView.onStart();
        super.onStart();
        requestMissingPermissions();
    }

    @Override
    protected void onResume() {
        super.onResume();
        playerView.onResume();
    }

    @Override
    protected void onPause() {
        playerView.onPause();
        super.onPause();
    }

    @Override
    protected void onStop() {
        playerView.onStop();
        super.onStop();
    }

    @Override
    protected void onDestroy() {
        // The BitmovinPlayer must be removed from the BitmovinPlayerNotificationManager before it is destroyed
        notificationManager.setPlayer(null);
        playerView.onDestroy();
        super.onDestroy();
    }

    protected void initializePlayer() {
        // Load a new source
        SourceConfig sourceConfig = SourceConfig.fromUrl("https://bitdash-a.akamaihd.net/content/sintel/sintel.mpd");
        sourceConfig.setPosterSource("https://bitmovin-a.akamaihd.net/content/sintel/poster.png");

        player.load(sourceConfig);
    }

    private final ActivityResultLauncher<String> requestPermissionLauncher = registerForActivityResult(
            new ActivityResultContracts.RequestPermission(),
            isGranted -> { /* Do nothing */ }
    );

    private void requestMissingPermissions() {
        if (Build.VERSION.SDK_INT < 33) return;

        if (ContextCompat.checkSelfPermission(this, Manifest.permission.POST_NOTIFICATIONS)
                != PackageManager.PERMISSION_GRANTED) {
            requestPermissionLauncher.launch(Manifest.permission.POST_NOTIFICATIONS);
        }
    }
}

File: BasicMediaControlKotlin/src/main/java/com/bitmovin/player/samples/mediacontrol/basic/MainActivity.kt
package com.bitmovin.player.samples.mediacontrol.basic

import android.media.AudioAttributes
import android.media.AudioFocusRequest
import android.media.AudioManager
import android.os.Build
import android.os.Bundle
import android.support.v4.media.session.MediaSessionCompat
import androidx.appcompat.app.AppCompatActivity
import com.bitmovin.player.PlayerView
import com.bitmovin.player.api.Player
import com.bitmovin.player.api.source.SourceConfig
import com.bitmovin.player.ui.notification.PlayerNotificationManager

class MainActivity : AppCompatActivity() {
    private val notificationChannelId = "com.bitmovin.player"
    private val notificationId = 1

    private lateinit var playerView: PlayerView
    private var player: Player? = null
    private lateinit var notificationManager: PlayerNotificationManager

    private lateinit var mediaSessionCompat: MediaSessionCompat

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        playerView = findViewById(R.id.playerView)
        player = playerView.player

        notificationManager = PlayerNotificationManager.createWithNotificationChannel(
                this,
                notificationChannelId,
                R.string.control_notification_channel,
                notificationId,
                null
        )
        notificationManager.setOngoing(true)
        notificationManager.setPlayer(player)

        initializePlayer()

        mediaSessionCompat = MediaSessionCompat(this, "Bitmovin Sample Media Session")
        notificationManager.setMediaSessionToken(mediaSessionCompat.sessionToken)

        requestAudioFocus()
    }

    override fun onDestroy() {
        // The BitmovinPlayer must be removed from the BitmovinPlayerNotificationManager before it is destroyed
        notificationManager.setPlayer(null)
        playerView.onDestroy()
        mediaSessionCompat.release()
        super.onDestroy()
    }

    override fun onStart() {
        playerView.onStart()
        super.onStart()
    }

    override fun onResume() {
        super.onResume()
        playerView.onResume()
    }

    override fun onPause() {
        playerView.onPause()
        super.onPause()
    }

    override fun onStop() {
        playerView.onStop()
        super.onStop()
    }

    private fun initializePlayer() {
        // Load a new source
        val sourceConfig = SourceConfig.fromUrl(
            "https://bitmovin-a.akamaihd.net/content/MI201109210084_1/mpds/f08e80da-bf1d-4e3d-8899-f0f6155f6efa.mpd"
        ).apply {
            posterSource = "https://bitmovin-a.akamaihd.net/content/poster/hd/RedBull.jpg"
            title = "Art of Motion"
            description = "Red Bull's Parkour event, this time in Santorini"
        }
        player?.load(sourceConfig)
    }

    private fun requestAudioFocus() {
        if (Build.VERSION.SDK_INT < 26) {
            requestAudioFocusBeforeApi26()
            return
        }

        val am = getSystemService(AUDIO_SERVICE) as AudioManager
        val focusRequest = AudioFocusRequest.Builder(AudioManager.AUDIOFOCUS_GAIN) // API 26
                .setAudioAttributes(
                        AudioAttributes.Builder() // API 21
                                .setUsage(AudioAttributes.USAGE_MEDIA)
                                .setContentType(AudioAttributes.CONTENT_TYPE_MOVIE)
                                .build()
                )
                .setAcceptsDelayedFocusGain(true)
                .setOnAudioFocusChangeListener { focusChange ->
                    // This listener is required or an exception will be thrown
                    // Default and expected behavior differs by Android versions. More details can be found at
                    // https://developer.android.com/guide/topics/media-apps/audio-focus

                    if (focusChange != AudioManager.AUDIOFOCUS_GAIN) {
                        player?.pause()
                    }
                }
                .build()

        val result = am.requestAudioFocus(focusRequest)

        if (result == AudioManager.AUDIOFOCUS_REQUEST_GRANTED) {
            // App has audiofocus now
            player?.play()
        }
    }

    private fun requestAudioFocusBeforeApi26() {
        val am = getSystemService(AUDIO_SERVICE) as AudioManager
        val result = am.requestAudioFocus(
                { focusChange ->
                    // onAudioFocusChangeListener needs to handle stopping/ducking media if focus is lost.
                    // More details can be found at https://developer.android.com/guide/topics/media-apps/audio-focus

                    if (focusChange != AudioManager.AUDIOFOCUS_GAIN) {
                        player?.pause()
                    }
                },
                AudioManager.STREAM_MUSIC,
                AudioManager.AUDIOFOCUS_GAIN
        )

        if (result == AudioManager.AUDIOFOCUS_REQUEST_GRANTED) {
            // App has audiofocus now
            player?.play()
        }
    }
}